imgproc: GCC13 warns -Wstringop-overflow at min_enclosing_convex_polygon.cpp

### System Information

OpenCV vesion: 4.x 95be2f2af8d70c1fa703192d9d338c53323b24a1
Operating System / Platform: Ubuntu 24.04
Compiler & compiler version: GCC 13.3.0

### Detailed description

With CMAKE_CXX_STANDARD=20, GCC13 warns `-Wstringop-overflow` at min_enclosing_convex_polygon.cpp.

This seems to be a false-positive warning from the compiler's static analysis.

>    inlined from ‘std::vector<minEnclosingConvexPolygon::Side> minEnclosingConvexPolygon::Chains::findKSides(int, int, int)’ at /home/kmtr/work/opencv4/modules/imgproc/src/min_enclosing_convex_polygon.cpp:920:24:
/usr/include/c++/13/bits/stl_construct.h:97:14: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]


### Steps to reproduce

```
cmake -S opencv4 -B build4-main.CXX20 -GNinja -DCMAKE_CXX_STANDARD=20
cmake build build4-main.CXX20
```

### Issue submission checklist

- [x] I report the issue, it's not a question
- [x] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [x] I updated to the latest OpenCV version and the issue is still there
- [x] There is reproducer code and related data files (videos, images, onnx, etc)

--- a/modules/imgproc/src/min_enclosing_convex_polygon.cpp
+++ b/modules/imgproc/src/min_enclosing_convex_polygon.cpp
@@ -913,6 +913,7 @@ std::vector<Side> Chains::reconstructHSidedChain(int h, int i, int j)
 std::vector<Side> Chains::findKSides(int k, int i, int j)
 {
     std::vector<Side> sides;
+    sides.reserve(k); // Optimization and avoiding GCC's false positive warning (-Wstringop-overflow)
     sides.push_back({i, true});
     sides.push_back({j, true});
 

